plugin/evm/limitorders/limit_order_tx_processor.go

```diff
@@ -141,18 +153,15 @@ func (lotp *limitOrderTxProcessor) executeLocalTx(contract common.Address, contr
 
 func (lotp *limitOrderTxProcessor) PurgeLocalTx() {
 	pending := lotp.txPool.Pending(true)
-	localAccounts := []common.Address{common.HexToAddress(userAddress1), common.HexToAddress(userAddress2)}
-
-	for _, account := range localAccounts {
-		if txs := pending[account]; len(txs) > 0 {
-			for _, tx := range txs {
-				_, err := getOrderBookContractCallMethod(tx, lotp.orderBookABI, lotp.orderBookContractAddress)
-				if err == nil {
-					lotp.txPool.RemoveTx(tx.Hash())
-				}
+	for _, txs := range pending {
+		for _, tx := range txs {
+			method, err := getOrderBookContractCallMethod(tx, lotp.orderBookABI, lotp.orderBookContractAddress)
+			if err == nil && method.Name == "executeMatchedOrders" {
```

Comment:
we need to remove `liquidate` and `settleFunding` txs too

Reply:
added